src/lib/ndpi_main.c

```diff
@@ -8448,6 +8526,8 @@ static ndpi_protocol ndpi_internal_detection_process_packet(struct ndpi_detectio
       return(ret);
 
     fpc_check_ip(ndpi_str, flow);
+    /* Update fpc cache */
+    ndpi_fpc_cache_update(ndpi_str,flow);
```

Comment:
Here we need to lookup into the cache, not update it!

Reply:
I will arrange the changes, are we going to update the cache in DNS dissector? or any other places? Please advice.